src/modules/trade/helpers/generate-trade.js

```diff
@@ -23,14 +23,17 @@ export const generateTrade = memoize((market, outcome, outcomeTradeInProgress, o
 
   const side = (outcomeTradeInProgress && outcomeTradeInProgress.side) || TRANSACTIONS_TYPES.BUY
   const numShares = (outcomeTradeInProgress && outcomeTradeInProgress.numShares) || null
+  const sharesFilled = (outcomeTradeInProgress && outcomeTradeInProgress.sharesFilled) || null
   const limitPrice = (outcomeTradeInProgress && outcomeTradeInProgress.limitPrice) || null
   const totalFee = (outcomeTradeInProgress && outcomeTradeInProgress.totalFee) || 0
+  const feePercent = (outcomeTradeInProgress && outcomeTradeInProgress.feePercent) || 0
   const gasFeesRealEth = (outcomeTradeInProgress && outcomeTradeInProgress.gasFeesRealEth) || 0
   const totalCost = (outcomeTradeInProgress && outcomeTradeInProgress.totalCost) || 0
-  const marketType = (market && market.type) || null
-  const minPrice = (market && market.minPrice) || null
+  const marketType = (market && market.marketType) || null
+  const minPrice = (market && typeof market.minPrice === 'number') ? market.minPrice : null
   const maxPrice = (market && market.maxPrice) || null
-  const preOrderProfitLoss = calcOrderProfitLossPercents(numShares, limitPrice, side, minPrice, maxPrice, marketType)
+  const adjustedTotalCost = (totalCost > 0) ? new BigNumber(outcomeTradeInProgress.totalCost, 10).minus(new BigNumber(outcomeTradeInProgress.totalFee, 10)).abs().toFixed() : null
```

Comment:
Just to double-check, is `totalCost` either a JS number or a BigNumber?  (i.e., can it be compared to 0 without converting it first?)